pkg/operator/mds/mds.go

```diff
@@ -41,151 +44,176 @@ const (
+	fs := &Filesystem{
+		ObjectMeta: metav1.ObjectMeta{Name: f.Name, Namespace: clusterName},
+		Spec: FilesystemSpec{
+			MetadataPool: pool.ModelToSpec(f.MetadataPool),
+			MetadataServer: MetadataServerSpec{
+				ActiveCount: f.MetadataServer.ActiveCount,
+			},
+		},
+	}
+	for _, p := range f.DataPools {
+		fs.Spec.DataPools = append(fs.Spec.DataPools, pool.ModelToSpec(p))
 	}
+
+	return fs.Create(context, version, hostNetwork)
 }
 
 // Start the mds manager
-func (c *Cluster) Start() error {
-	logger.Infof("start running mds")
+func (f *Filesystem) Create(context *clusterd.Context, version string, hostNetwork bool) error {
+	if err := f.validate(); err != nil {
+		return err
+	}
+
+	var dataPools []*model.Pool
+	for _, pool := range f.Spec.DataPools {
+		dataPools = append(dataPools, pool.ToModel(""))
+	}
+	fs := cephmds.NewFS(f.Name, f.Spec.MetadataPool.ToModel(""), dataPools, f.Spec.MetadataServer.ActiveCount)
+	if err := fs.CreateFilesystem(context, f.Namespace); err != nil {
+		return fmt.Errorf("failed to create file system %s: %+v", f.Name, err)
+	}
 
-	id := "mds1"
-	err := c.createKeyring(c.context.Clientset, id)
+	filesystem, err := client.GetFilesystem(context, f.Namespace, f.Name)
 	if err != nil {
-		return fmt.Errorf("failed to create mds keyring. %+v", err)
+		return fmt.Errorf("failed to get file system %s. %+v", f.Name, err)
 	}
 
+	logger.Infof("start running mds for file system %s", f.Name)
+
 	// start the deployment
-	deployment := c.makeDeployment(id)
-	_, err = c.context.Clientset.ExtensionsV1beta1().Deployments(c.Namespace).Create(deployment)
+	deployment := f.makeDeployment(strconv.Itoa(filesystem.ID), version, hostNetwork)
+	_, err = context.Clientset.ExtensionsV1beta1().Deployments(f.Namespace).Create(deployment)
 	if err != nil {
 		if !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("failed to create mds deployment. %+v", err)
 		}
-		logger.Infof("mds deployment already exists")
+		logger.Infof("mds deployment %s already exists", deployment.Name)
 	} else {
-		logger.Infof("mds deployment started")
+		logger.Infof("mds deployment %s started", deployment.Name)
 	}
 
 	return nil
 }
 
-func (c *Cluster) createKeyring(clientset kubernetes.Interface, id string) error {
-	_, err := clientset.CoreV1().Secrets(c.Namespace).Get(appName, metav1.GetOptions{})
-	if err == nil {
-		logger.Infof("the mds keyring was already generated")
-		return nil
-	}
-	if !errors.IsNotFound(err) {
-		return fmt.Errorf("failed to get mds secrets. %+v", err)
-	}
-
-	// get-or-create-key for the user account
-	keyring, err := cephmds.CreateKeyring(c.context, c.Namespace, id)
-	if err != nil {
-		return fmt.Errorf("failed to create mds keyring. %+v", err)
+// Start the mds manager
+func (f *Filesystem) Delete(context *clusterd.Context) error {
+	// Delete the deployment/daemonset
+	k8sutil.DeleteDaemonDeployment(context, f.Namespace, f.instanceName())
```

Comment:
it seems we are only creating a deployment for mds. Why are we also trying to delete Daemonset?

Reply:
actually i can simplify this now. i started out with a daemonset in the design, but now it's only a deployment